File: 251127/요일 맞추기/guess-day-of-week.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const [m1, d1, m2, d2] = input[0].split(' ').map(Number);
// Please Write your code here.

const days = [0,31,28,31,30,31,30,31,31,30,31,30,31];
const res = ['Mon','Tue','Wed','Thu','Fri','Sat','Sun']

function getDays(m,d){
    let sum=0;
    for(let i=1; i<m; i++){
        sum+=days[i];
    }
    sum+=d;
    return sum;
}

let day = getDays(m2,d2)-getDays(m1,d1);

while(day<0){
    day+=7;
}

console.log(res[day%7])

File: 260523/요일 맞추기/guess-day-of-week.cpp
#include <iostream>
#include <string>
using namespace std;

int getDays(int m,int d){
    int days[13] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int totalDay = 0;
    for(int i=1; i<m; i++){
        totalDay+=days[i];
    }
    totalDay+=d;

    return totalDay;
}
int main() {
    string days[7] = { "Mon", "Tue", "Wed", "Thu", "Fri", "Sat","Sun"};


    int m1,d1,m2,d2;
    cin >>m1>>d1>>m2>>d2;

    int sub =getDays(m2,d2)- getDays(m1,d1) ;

    sub%=7;
    if(sub<0) sub+=7;
    cout<<days[sub];

    return 0;
}